File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;
class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;

        List<String> cache = new ArrayList<>();

        for (String city : cities) {
            city = city.toLowerCase();
            int index = cache.indexOf(city);

            if (index >= 0) {
                answer += 1;
                cache.remove(index);
                cache.add(city);
            } else {
                answer += 5;
                cache.add(city);
            }

            if (cache.size() > cacheSize) {
                cache.remove(0);
            }

        }
        return answer;
    }
}

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.js
function solution(cacheSize, cities) {
  let time = 0;
  const cache = [];
  for (let city of cities) {
    city = city.toLowerCase();
    const index = cache.indexOf(city);
    if (index === -1) {
      cache.push(city);
      time += 5;
    } else {
      cache.splice(index, 1);
      cache.push(city);
      time += 1;
    }
    if (cache[cacheSize]) {
      cache.shift();
    }
  }

  return time;
}